### ISSUE TITLE
Adding Lovász-Softmax loss to the binary cross-entropy loss for segmentation

### ISSUE THREAD

**Issue by genji970** [ultralytics/ultralytics#20862]
### Search before asking

- [x] I have searched the Ultralytics [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar feature requests.

### Description

utils/loss.py, 392th line, when we look at 'single_mask_loss', 

previous
```python

pred_mask = torch.einsum("in,nhw->ihw", pred, proto)  # (n, 32) @ (32, 80, 80) -> (n, 80, 80)
loss = F.binary_cross_entropy_with_logits(pred_mask, gt_mask, reduction="none")
return (crop_mask(loss, xyxy).mean(dim=(1, 2)) / area).sum()

```
changed

```python

pred_mask = torch.einsum("in,nhw->ihw", pred, proto)  # (n, 32) @ (32, 80, 80) -> (n, 80, 80)

lovasz_loss = torch.stack([
    lovasz_hinge(pred_mask[i], gt_mask[i])
    for i in range(pred_mask.size(0))
]).mean()

loss = F.binary_cross_entropy_with_logits(pred_mask, gt_mask, reduction="none") + lambda * lovasz_loss

return (crop_mask(loss, xyxy).mean(dim=(1, 2)) / area).sum()
```

It will take more to calculate loss but lovasz loss makes model to enhance boundary prediction accuracy. So what about making this loss as optional?

### Are you willing to submit a PR?

- [x] Yes I'd like to help by submitting a PR!

**Comment by genji970** [comment]
@glenn-jocher thank you for your response. okay. I also think it is way better than previous one. I will open pr and put three considerations. Thank you!

**Comment by glenn-jocher** [comment]
Great! Looking forward to your PR. When implementing, you might want to reference how other optional loss components are handled in the codebase - for example, the DFL loss in the `BboxLoss` class uses a conditional check with `self.dfl_loss` to only compute when needed. This pattern could work well for the Lovász loss too.
